app/app.rb

```diff
@@ -0,0 +1,66 @@
+require "sinatra/base"
+require "sinatra/json"
+require "./lib/menu"
+require "./lib/order_list"
+require "./lib/order"
+require "./lib/tax"
+require "./lib/discount"
+require "./lib/receipt"
+require "./lib/payment"
+require "./app/modules/my_helpers"
+
+class TillTechTest < Sinatra::Base
+  set :root, File.dirname(__FILE__)
+  alias_method :s, :settings
+  helpers Sinatra::JSON
+  include MyHelpers
+  set :static, true
+  set :order_list, OrderList.new
+  set :menu,       Menu.new([{ name: "Cafe Latte", price: 2.5 },
+                             { name: "Cappucino",  price: 3.0 },
+                             { name: "Flat White", price: 2.3 },
+                             { name: "Lasagne",    price: 5.0 },
+                             { name: "Risotto",    price: 4.5 },
+                             { name: "Tiramisu",   price: 3.6 },
+                             { name: "Muffin",     price: 3.8, discount: "5%" }
+                            ])
+  set :utilities, tax: Tax.new("8.64%"),
+             discount: Discount.new(discount: "10%", 
+                               discountable?: proc { |value| value > 30 })
```

Comment:
Do you really need to have this as a proc that you pass in in the constructor? Does the discount value actually change in the object? Unless you actually need something that can take different values like this then don't add more than you need. Also the discount value should probably be a float rather than a string